### DISCUSSION TITLE
Is there a way to get rid of /api when using next auth with route handlers

### DISCUSSION THREAD

**Discussion by VinhTan** [nextauthjs/next-auth#8582] [Help]
I can see this note in nextauth documentation:

> Technically, in a Route Handler, the api/ prefix is not necessary, but we decided to keep it required for an easier migration.

but i'm still wondering if there's an option or a futur feature to get rid of this requirement ?

**Comment by uchar** [comment]
same problem here , we use /api for our main api that is handled by nginx  and now we need to completely change that because of this, prefixing next auth with a constant magical string is very weird

**Comment by Dragate** [comment]
[`basePath`](https://authjs.dev/reference/core#basepath) option should be what you're looking for.
